frontend/src/features/entry/pages/EntryNamePage/EntryNamePage.tsx

```diff
@@ -3,24 +3,37 @@ import Button from '@/components/@common/Button/Button';
 import Headline3 from '@/components/@common/Headline3/Headline3';
 import Input from '@/components/@common/Input/Input';
 import ProgressCounter from '@/components/@common/ProgressCounter/ProgressCounter';
+import { usePlayerRole } from '@/contexts/PlayerRoleContext';
 import Layout from '@/layouts/Layout';
 import { useState } from 'react';
-import { useNavigate } from 'react-router-dom';
+import { useLocation, useNavigate } from 'react-router-dom';
 import * as S from './EntryNamePage.styled';
 
 const MAX_NAME_LENGTH = 10;
 
 const EntryNamePage = () => {
   const [name, setName] = useState('');
   const navigate = useNavigate();
+  const { state } = useLocation();
+  const { playerRole } = usePlayerRole();
 
   const handleNavigateToHome = () => navigate('/');
-  const handleNavigateToMenu = () =>
-    navigate('/entry/menu', {
-      state: {
-        name,
-      },
-    });
+  const handleNavigateToMenu = () => {
+    if (playerRole === 'HOST') {
+      navigate('/entry/menu', {
+        state: {
+          name,
+        },
+      });
+    } else if (playerRole === 'GUEST') {
+      navigate('/entry/menu', {
+        state: {
+          name,
+          joinCode: state.joinCode,
+        },
+      });
+    }
+  };
```

Comment:
[P5] 그냥 제 개인적인 취향이긴한데, 여러줄로 되어있으니까 조금 난잡해보여서,, 이렇게 하는건 어때염 ㅎㅎ prettier가 허용 해준다면..

```suggestion
  const handleNavigateToMenu = () => {
    if (playerRole === 'HOST') {
      navigate('/entry/menu', { state: { name } });
    } else if (playerRole === 'GUEST') {
      navigate('/entry/menu', { state: { name, joinCode: state.joinCode } });
    }
  };
```